internal/agent/reset.go

```diff
@@ -140,26 +140,25 @@ func sharedReset(reboot, unattended, resetOem bool, dir ...string) (c *config.Co
 	bus.Manager.Publish(sdk.EventBeforeReset, sdk.EventPayload{}) //nolint:errcheck
 
 	// Prepare a config from the cli flags
-	r := map[string]map[string]interface{}{
-		"reset": {},
-	}
+	r := ExtraConfigReset{}
+	r.Reset.ResetOem = resetOem
 
 	if resetOem {
-		r["reset"]["reset-oem"] = "true"
+		r.Reset.ResetOem = true
 	}
 
 	if reboot {
-		r["reset"]["reboot"] = reboot
+		r.Reset.Reboot = reboot
```

Comment:
wait I just noticed, left side is boolean and right side is string, so probably also needs something like in 147 or like in 158

Reply:
? reboot is boolean no?

Reply:
doesn't it come from the argument list in line 92? but indeed with the change it should work eitherway